exercises/react/react.ts

```diff
@@ -0,0 +1,102 @@
+//
+// Delete and replace stub with your own implementation
+//
+// Inspired by "How it Works":
+// https://indepth.dev/posts/1269/finding-fine-grained-reactive-programming#how-it-works
+// https://levelup.gitconnected.com/finding-fine-grained-reactive-programming-89741994ddee?source=friends_link&sk=31c66a70c1dce7dd5f3f4229423ad127#4543
+//
+// and "Computations":
+// https://github.com/ryansolid/solid/blob/master/documentation/reactivity.md#user-content-computations
+//
+
+type EqualFn<T> = (lhs: T, rhs: T) => boolean
+type GetterFn<T> = () => T
+type SetterFn<T> = (v: T) => T
+type UnsubscribeFn = () => void
+type UpdateFn<T> = (value?: T) => T
+
+type InputCouple<T> = [GetterFn<T>, SetterFn<T>]
+
+type Options = {
+  name: string // for debugging
+}
+
+type ObserverR = {
+  name?: string
+}
+
+type ObserverV<T> = {
+  value?: T
+  fn: UpdateFn<T>
+}
+
+type Observer<T> = ObserverR & ObserverV<T>
+
+type SubjectR = {
+  name?: string
+  observer: ObserverR | undefined
+}
+
+type SubjectV<T> = {
+  value: T
+  equal?: EqualFn<T>
```

Comment:
```suggestion
  equalFn?: EqualFn<T>
```

Just so it's in-line with `ObserverV`